src/main/java/org/apache/cassandra/sidecar/data/StreamSSTableComponentRequest.java

```diff
@@ -43,24 +45,74 @@ public class StreamSSTableComponentRequest extends SSTableComponent
     {
-        this(new QualifiedTableName(keyspace, tableName, true), snapshotName, componentName);
+        this(new QualifiedTableName(keyspace, tableName, true), snapshotName, null, componentName);
+    }
+
+    /**
+     * Constructor for the holder class
+     *
+     * @param keyspace      the keyspace in Cassandra
+     * @param tableName     the table name in Cassandra
+     * @param snapshotName  the name of the snapshot
+     * @param indexName     the name of the index for the SSTable component
+     * @param componentName the name of the SSTable component
+     */
+    @VisibleForTesting
+    public StreamSSTableComponentRequest(String keyspace, String tableName, String snapshotName, String indexName,
+                                         String componentName)
+    {
+        this(new QualifiedTableName(keyspace, tableName, true), snapshotName, indexName, componentName);
+    }
+
+    /**
+     * Constructor for the holder class
+     *
+     * @param qualifiedTableName the qualified table name in Cassandra
+     * @param snapshotName       the name of the snapshot
+     * @param componentName      the name of the SSTable component
+     */
+    public StreamSSTableComponentRequest(QualifiedTableName qualifiedTableName,
+                                         String snapshotName,
+                                         String componentName)
+    {
+        this(qualifiedTableName, snapshotName, null, componentName);
     }
 
     /**
      * Constructor for the holder class
      *
      * @param qualifiedTableName the qualified table name in Cassandra
      * @param snapshotName       the name of the snapshot
+     * @param indexName          the name of the index for the SSTable component
      * @param componentName      the name of the SSTable component
      */
     public StreamSSTableComponentRequest(QualifiedTableName qualifiedTableName,
                                          String snapshotName,
+                                         @Nullable String indexName,
                                          String componentName)
     {
-        super(qualifiedTableName, componentName);
+        super(qualifiedTableName, indexName, componentName);
         this.snapshotName = Objects.requireNonNull(snapshotName, "snapshotName must not be null");
     }
 
+    public static StreamSSTableComponentRequest from(QualifiedTableName qualifiedTableName, RoutingContext context)
+    {
+        String snapshotName = context.pathParam("snapshot");
+        String indexName = context.pathParam("index");
+        String componentName = context.pathParam("component");
+
+        if (indexName == null)
+        {
+            return new StreamSSTableComponentRequest(qualifiedTableName,
+                                                     snapshotName,
+                                                     componentName);
+        }
```

Comment:
The nullness check seems unnecessary. The `StreamSSTableComponentRequest` allows taking null value.